File: vortexdbg-android/src/main/java/com/vortexdbg/linux/android/dvm/array/PrimitiveArray.java
package com.vortexdbg.linux.android.dvm.array;

import com.vortexdbg.Emulator;
import com.vortexdbg.linux.android.dvm.Array;
import com.vortexdbg.pointer.VortexdbgPointer;
import com.sun.jna.Pointer;

public interface PrimitiveArray<T> extends Array<T> {

    VortexdbgPointer _GetArrayCritical(Emulator<?> emulator, Pointer isCopy);

    void _ReleaseArrayCritical(Pointer elems, int mode);

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/android/dvm/array/PrimitiveArray.kt
package com.vortexdbg.linux.android.dvm.array

import com.vortexdbg.Emulator
import com.vortexdbg.linux.android.dvm.Array
import com.vortexdbg.pointer.VortexdbgPointer
import com.sun.jna.Pointer

interface PrimitiveArray<T> : Array<T> {

    fun _GetArrayCritical(emulator: Emulator<*>, isCopy: Pointer?): VortexdbgPointer

    fun _ReleaseArrayCritical(elems: Pointer, mode: Int)

}
